helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/AuthenticationSteps.java

```diff
@@ -0,0 +1,141 @@
+package org.symphonyoss.symphony.bots.helpdesk.bot.it.steps;
+
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.client.ClientProperties;
+import org.jbehave.core.annotations.When;
+import org.springframework.stereotype.Component;
+import org.symphonyoss.client.SymphonyClient;
+import org.symphonyoss.client.SymphonyClientConfig;
+import org.symphonyoss.client.SymphonyClientConfigID;
+import org.symphonyoss.client.exceptions.AuthenticationException;
+import org.symphonyoss.client.exceptions.InitException;
+import org.symphonyoss.client.impl.SymphonyBasicClient;
+import org.symphonyoss.client.model.SymAuth;
+import org.symphonyoss.symphony.bots.helpdesk.bot.authentication.HelpDeskAuthenticationException;
+import org.symphonyoss.symphony.bots.helpdesk.bot.config.HelpDeskBotConfig;
+import org.symphonyoss.symphony.bots.helpdesk.bot.config.HttpClientConfig;
+import org.symphonyoss.symphony.bots.helpdesk.bot.it.TestContext;
+import org.symphonyoss.symphony.clients.AuthenticationClient;
+import org.symphonyoss.symphony.pod.invoker.JSON;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+
+/**
+ * Class responsible for managing authentication steps.
+ * <p>
+ * Created by rsanchez on 23/02/18.
+ */
+@Component
```

Comment:
Is the annotation required?

Reply:
Yes, it's required to support dependency injection of HelpDeskConfig class and then get the YAML settings.